conda/auxlib/entity.py

```diff
@@ -972,19 +973,4 @@ def update(self, E=None, **F):
             self[k] = F[k]
 
 
-class EntityEncoder(JSONEncoder):
-    # json.dumps(obj, cls=SetEncoder)
-    def default(self, obj):
-        if hasattr(obj, 'dump'):
-            return obj.dump()
-        elif hasattr(obj, '__json__'):
-            return obj.__json__()
-        elif hasattr(obj, 'to_json'):
-            return obj.to_json()
-        elif hasattr(obj, 'as_json'):
-            return obj.as_json()
-        elif isinstance(obj, Enum):
-            return obj.value
-        elif isinstance(obj, Path):
-            return str(obj)
-        return JSONEncoder.default(self, obj)
+deprecated.constant("25.9", "26.3", "EntityEncoder", CondaJSONEncoder, addendum="Use `conda.common.serialize.json.CondaJSONEncoder` instead.",)
```

Comment:
The title of this PR was misleading. This is not monkeypatching since we are not setting properties on the global json module. It is subclassing and using a [standard json module feature.](https://docs.python.org/3/library/json.html#json.dump)

This does seem like a slow function. Would a faster version keep a mapping of `type(obj) -> (needs custom handling)` to avoid doing hasattr all the time.

Or we continue to move away from auxlib and don't do any of this.